### ISSUE TITLE
HELP WANTED: Translate Padloc to other languages 

### ISSUE THREAD

**Issue by MaKleSoft** [padloc/padloc#248]
Title says it all. The Padloc app is currently available in English and German, but we'd love to support more languages! If you'd like to help translate Padloc into your language (or any language you are sufficiently fluent in), here is how you can do it:

1. Fork this repo
2. Create a copy of the [empty translations template](../blob/master/packages/locale/res/translations/_template.json)
3. Change the name to the appropriate [country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements)
4. Fill in the translations
5. Create a pull request

Feel free to use this thread to let others know you're working on a translation or to coordinate your efforts.

That's it!  For more information, check out the [locale package](../blob/master/packages/locale/). Looking forward to your contributions 😊

**Comment by moigarmo** [comment]
Hi guys! I'm happy to help here. 
I'm translating into spanish.

Cheers.

**Comment by MaKleSoft** [comment]
@moigarmo Great! Let me know if you have any questions.

**Comment by moigarmo** [comment]
I've created a pull request with the translations in spanish.
I hope it helps.

**Comment by MaKleSoft** [comment]
@moigarmo Wow, that was fast. Thanks!

**Comment by moigarmo** [comment]
Hi @MaKleSoft, I've been playing around with the app a bit and I found a couple of errors in the translations I'd like to fix. But I don't know how to handle one kind of them. 
In spanish there are some sentences that in english are written the same but depending on the context they may have a different meaning and because of the way translations work in this project it is not easy to differentiate between them.
For instance, "log in" in the main screen means that you have to log in to perform further actions, whereas when you are creating a new item, the option log in means that you want to create a log in information item, which would be translated a bit diffetent.
I don't think it's a big deal but it seems awkard the way it is displayed in that screen. And the way translations work in this project don't allow, that I know of, a way to translate the same sentence differently.

If you know a way to do that it'd be great.

Thanks,
Moi.

**Comment by MaKleSoft** [comment]
@moigarmo Yes, I've noticed the same issue with German, specifically about the different meanings of "Login". Would you mind creating a separate issue listing all problems and where in the app they occur (screenshots would be amazing)? I'll figure out a strategy based on that.

**Comment by avvolodin** [comment]
I've added PR for russian ui translation.
[https://github.com/padloc/padloc/pull/258](https://github.com/padloc/padloc/pull/258)

**Comment by jhjacobs81** [comment]
I’ll start on the Dutch translation :)

**Comment by fghzxm** [comment]
What country code should I use for Simplified Chinese?  Should I use `zh-cn` or just `cn`?

**Comment by MaKleSoft** [comment]
Hi @fghzxm! First of all, **thank you** for taking the time to translate Padloc to Simplified Chinese! However, before you get started, I'd like to point out that we're currently working full steam on v4, which will have quite a few changes, including translatable texts - so if you want your contribution to last for more than a couple of weeks, it would be great if you could base your translation of the [`v4` branch](https://github.com/padloc/padloc/tree/v4). As for which locale code to use, I'm not entire sure honestly, but I think `cn` should be fine. It kind of depends on what locale string the user agents spits out, so we'll likely just have to test it.

**Comment by coluzziandrea** [comment]
I've added a [PR ](https://github.com/padloc/padloc/pull/520) for italian translations, hope it helps

**Comment by MaKleSoft** [comment]
FYI, translation files have been updated to include the latest changes from v4.

**Comment by rhuba8324** [comment]
I've added PR for Chinese Traditional (tw) translation. :D 
#547

**Comment by andrejdaskalov** [comment]
Started working on Macedonian translation. Also i have a question: in some places where every word is capitalized, is it important that the translated version is also capitalized that way? For example: "Enter Email Address"

**Comment by BrunoBernardino** [comment]
Thanks @andrejdaskalov ! Not all translations respect that, but it would be better for consistency, yes.

**Comment by WizMe-M** [comment]
Hi, currently am updating RU-translation (prev version: #258).
Could somebody tell me how can I test it out before pull request (does it available to)?

**Comment by BrunoBernardino** [comment]
@WizMe-M locally, assuming you've ran `npm ci` to get the dependencies installed, you can run `npm run format && npm test` to check locally before pushing or opening a PR.

**Comment by RoyalCrafter** [comment]
Hi, just to let you know, i'm currently updating the german translation, so that it includes the strings from the new padloc version

**Comment by MahmoudTurkmani** [comment]
Hello, I've begun work on an Arabic translation :)

**Comment by MahmoudTurkmani** [comment]
Create PR for Arabic translation #743 

Please note that some words that are used in the system don't have an Arabic equivalent or their equivalent is not suitable for use (such as uppercase, item, theme, and session). I've tried my best to pick the most suitable replacement for each of these words.

**Comment by MinionAttack** [comment]
Hi, I have updated and fixed Spanish translation. Here is the pull request: https://github.com/padloc/padloc/pull/744

**Comment by vhespanha** [comment]
Hey!!

Will begin the Portuguese translation today.

**Comment by vhespanha** [comment]
Just finished the Brazilian Portuguese translation, since some sentences required a little bit more context because of the differences between the two languages, I'm considering this a draft. Once it's merged, I'll make sure to use it in the app and see if i can spot any improvements.

[#756](https://github.com/padloc/padloc/pull/756) 😁

**Comment by MahmoudTurkmani** [comment]
Hi, the Arabic translation PR has been done for quite a while just pending merge I think #743

**Comment by vhespanha** [comment]
Yeah, no commits in 7 months either.
